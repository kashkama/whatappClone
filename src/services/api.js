import { initialize, setListener, pushData } from "./firebase";

const mockMessages = [
    {
        incoming: true,
        message: "Hi Owen"
    },
    {
        incoming: false,
        message: "Hi, Vader"
    },
    {
        incoming: true,
        message: "When we will learn real data fetching?"
    },
    {
        incoming: false,
        message: "First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?"
    }
];

export const getMockData = () => (
    new Promise(resolve => setTimeout(() => resolve(mockMessages), 1000))
)

export const initApi = () => initialize();

export const getMessages = (updaterFn) => setListener("messages", updaterFn);

export const postMessage = (message) => {
    if(Boolean(message)) {
        pushData("messages", {
            incoming: false,
            message
        })
    }
};