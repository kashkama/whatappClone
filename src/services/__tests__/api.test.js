import "react-native";
import React from "react";
import {getMockData} from "./../api";

it("returns an array with data", async () => {
    const messages = await getMockData();
    expect(messages).toMatchSnapshot();
});