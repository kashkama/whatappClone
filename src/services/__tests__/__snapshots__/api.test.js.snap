// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns an array with data 1`] = `
Array [
  Object {
    "incoming": true,
    "message": "Hi Owen",
  },
  Object {
    "message": "Hi, Purity",
    "outgoing": true,
  },
  Object {
    "incoming": true,
    "message": "When we will learn real data fetching?",
  },
  Object {
    "message": "First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?",
    "outgoing": true,
  },
]
`;
